### DISCUSSION TITLE
Constructable Stylesheets and LitElement

### DISCUSSION THREAD

**Discussion by nemethmik** [lit/lit#2220] [Q&A] [answered]
Hello to the Lit Community! 
I have recently found this article [](https://developers.google.com/web/updates/2019/02/constructable-stylesheets#constructing_a_stylesheet) and I was wondering, if it is a more efficient way of sharing big block of CSS between light and shadow DOM than what is suggested in the [Sharing Styles section in the Lit documentation](https://lit.dev/docs/components/styles/#sharing-styles)?
Have you tried it or considered giving it a try?

**Comment by justinfagnani** [comment, marked as answer]
The `css` template tag already uses constructable Stylesheets where available. You can also pass a CSSStyleSheet object directly to element styles if you construct one on your own.

**Reply by nemethmik** [reply]
Thank you so much, I've just found the API [adoptStyles](https://lit.dev/docs/api/styles/#adoptStyles)

**Reply by justinfagnani** [reply]
You don't need to use that directly you can do:

```ts
const stylesheet = new CSSStyleSheet();
stylesheet.replaceSync(`...`);

class MyElement extends LitElement {
  styles = [stylesheet];
}
```

**Reply by nemethmik** [reply]
Awesome, thank you so much, I am only on Lit for only about a month, and I am still amazed how brilliantly engineered this toolset is.

**Reply by nemethmik** [reply]
Am I right when I say that **unsafeCSS** is roughly equivalent to using *CSSStyleSheet* and *replaceSync*?
